src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
+    ComponentRegister.init({});
+
+    expect(ComponentRegister.Link).toStrictEqual("a");
+  });
+});
+it("registers link by default as an anchor tag", () => {
+  expect(
+    <ComponentRegister.Link href="/"> Test</ComponentRegister.Link>
+  ).toStrictEqual(<a href="/"> Test</a>);
```

Comment:
If this test were appropriate, it would probably be better as a snapshot. It makes it easier to update. See above for why I think this isn't the appropriate test in this instance, though.